### ISSUE TITLE
Asset.getDisplayValue only correct where 18 decimal places is the token precision

### ISSUE THREAD

**Issue by aaronmboyd** [austintgriffith/burner-core#1]
```
getDisplayValue = (value, decimals=2) => {
    const displayVal = utils.fromWei(value.toString(), 'ether');
    if (displayVal.indexOf('.') !== -1) {
      return displayVal.substr(0, displayVal.indexOf('.') + decimals + 1);
    }
    return displayVal;
}
```

By using utils.fromWei it implies that the base unit is always 10^18 away from 1. While it isn't part of the `IERC20.js` interface ABI, it's very common for tokens to use the ERC20Detailed definition (https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20Detailed.sol) which may specify a different number of decimals than 18.

Suggest maybe update ERC20Asset, add a new optional parameter; decimals and do some fancy BigNumber mathematics to display the token appropriately rather than `fromWei`. Alternatively, create a new subclass ERC20AssetDetailed which can implement this functionality.

A cool future way might be to include the ERC20AssetDetailed ABI and actually call the `decimals()` function to determine appropriate decimals on construction of the Asset.

**Comment by dmihal** [comment]
Do you know of many tokens that don't use 18 decimal places? There seems to be some consensus in the community that all tokens should use 18 (See [EIP724](https://github.com/ethereum/EIPs/issues/724)).

I'd like to avoid adding too much complexity to places that don't need it, give that every token I can find uses 18 decimal places.

**Comment by aaronmboyd** [comment]
Telcoin (TEL) is one that I can think of immediately (two decimal places), but I do realise it's not very common. Isn't the idea of subclassing ERC20Asset so that it can have slightly different behaviour to a native Asset? Maybe if I want it I should subclass ERC20Asset and add in the _detailed_ ABI myself, and see what you think.

**Comment by dmihal** [comment]
Actually one easy way would be to subclass ERC20Asset. Here's an example that does that for 4 decimal places:

```
class ERC20Asset4Decimals extends ERC20Asset {
  getDisplayValue(value, decimals=2) {
    const displayVal = fromWei(value.toString(), 'kwei');
    if (displayVal.indexOf('.') !== -1) {
      return displayVal.substr(0, displayVal.indexOf('.') + decimals + 1);
    }
    return displayVal;
  }
}
```

A more elegant solution would be to let the number of decimals be an optional argument for the constructor, I'll look into how easy it would be to do that conversion.

**Comment by dmihal** [comment]
Ok so I think i've implemented this: any asset will now accept `decimals` in the constructor (defaulting to 18).

You can see the change in aa3beec59e66bdf20f13a9f2251e4a847514dba6, and it's released in @burner-wallet/assets@0.0.18.

I'm closing this as I think it's resolved the issue, but let me know if there's any issues with this!
